diff --git a/lib/Drupal/Core/Asset/AssetDumper.php b/lib/Drupal/Core/Asset/AssetDumper.php
index c01b5f3b63..f68a7caba4 100644
--- a/lib/Drupal/Core/Asset/AssetDumper.php
+++ b/lib/Drupal/Core/Asset/AssetDumper.php
@@ -40,7 +40,8 @@ public function dump($data, $file_extension) {
     $path = 'assets://' . $file_extension;
     // Prefix filename to prevent blocking by firewalls which reject files
     // starting with "ad*".
-    $filename = $file_extension . '_' . Crypt::hashBase64($data) . '.' . $file_extension;
+    $base_path = base_path();
+    $filename = $file_extension . '_' . Crypt::hashBase64($data . $base_path) . '.' . $file_extension;
     $uri = $path . '/' . $filename;
     return $this->dumpToUri($data, $file_extension, $uri);
   }
diff --git a/lib/Drupal/Core/Asset/AssetResolver.php b/lib/Drupal/Core/Asset/AssetResolver.php
index fcd294a649..b6219d4494 100644
--- a/lib/Drupal/Core/Asset/AssetResolver.php
+++ b/lib/Drupal/Core/Asset/AssetResolver.php
@@ -143,7 +143,8 @@ public function getCssAssets(AttachedAssetsInterface $assets, $optimize, ?Langua
     $theme_info = $this->themeManager->getActiveTheme();
     // Add the theme name to the cache key since themes may implement
     // hook_library_info_alter().
-    $cid = 'css:' . $theme_info->getName() . ':' . $language->getId() . Crypt::hashBase64(serialize($libraries_to_load)) . (int) $optimize;
+    $base_path = base_path();
+    $cid = 'css:' . $theme_info->getName() . ':' . $language->getId() . Crypt::hashBase64(serialize($libraries_to_load) . $base_path) . (int) $optimize;
     if ($cached = $this->cache->get($cid)) {
       return $cached->data;
     }

diff --git a/lib/Drupal/Core/Asset/CssCollectionOptimizer.php b/lib/Drupal/Core/Asset/CssCollectionOptimizer.php
index 94712188b..2f9bb9914 100644
--- a/lib/Drupal/Core/Asset/CssCollectionOptimizer.php
+++ b/lib/Drupal/Core/Asset/CssCollectionOptimizer.php
@@ -174,7 +174,8 @@ protected function generateHash(array $css_group) {
     foreach ($css_group['items'] as $css_file) {
       $css_data[] = $css_file['data'];
     }
-    return hash('sha256', serialize($css_data));
+    $base_path = base_path();
+    return hash('sha256', serialize($css_data) . $base_path);
   }

   /**