diff --git a/modules/system/src/Form/PerformanceForm.php b/modules/system/src/Form/PerformanceForm.php
index 7b3890961a..78cb66876a 100644
--- a/modules/system/src/Form/PerformanceForm.php
+++ b/modules/system/src/Form/PerformanceForm.php
@@ -155,6 +155,12 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       '#config_target' => 'system.performance:js.preprocess',
       '#disabled' => $disabled,
     ];
+    $form['bandwidth_optimization']['stale_file_threshold'] = [
+      '#type' => 'textfield',
+      '#title' => t('Aggregate stale files expire time in second'),
+      '#config_target' => 'system.performance:stale_file_threshold',
+      '#disabled' => $disabled,
+    ];
 
     return parent::buildForm($form, $form_state);
   }
