diff --git a/modules/lightning_media_bulk_upload/src/Form/BulkUploadForm.php b/modules/lightning_media_bulk_upload/src/Form/BulkUploadForm.php
index 729173741..31ae35099 100644
--- a/modules/lightning_media_bulk_upload/src/Form/BulkUploadForm.php
+++ b/modules/lightning_media_bulk_upload/src/Form/BulkUploadForm.php
@@ -148,6 +148,7 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
         'query' => [
           'bulk_create' => implode(',', $bulk_create),
         ],
+        'language' => \Drupal::languageManager()->getCurrentLanguage()
       ]);
       $form_state->setRedirectUrl($redirect);
     }
diff --git a/src/Form/MediaForm.php b/src/Form/MediaForm.php
index 2051bd6e0..ad0405d51 100644
--- a/src/Form/MediaForm.php
+++ b/src/Form/MediaForm.php
@@ -134,7 +134,9 @@ public function save(array $form, FormStateInterface $form_state) {
     $next_entity_id = array_shift($queue);
     $next_entity_edit_form = $this->entityTypeManager->getStorage($entity->getEntityTypeId())
       ->load($next_entity_id)
-      ->toUrl('edit-form');
+      ->toUrl('edit-form', [
+        'language' => \Drupal::languageManager()->getCurrentLanguage()
+      ]);
 
     // If there are more entities to edit, ensure they're mentioned in the query
     // string of the next entity's edit form.
