diff --git a/src/Form/UserDefaultPageConfigEntityForm.php b/src/Form/UserDefaultPageConfigEntityForm.php
index bd7651a..9a5f9ca 100644
--- a/src/Form/UserDefaultPageConfigEntityForm.php
+++ b/src/Form/UserDefaultPageConfigEntityForm.php
@@ -254,15 +254,15 @@ class UserDefaultPageConfigEntityForm extends EntityForm {
     $input = $form_state->getUserInput();
     $user_default_page_config_entity = $this->entity;
     $user_input = $input['users'];
+    $users_array = '';
     if (!empty($user_input)) {
       $uids = explode(',', $user_input);
-      $users_array = '';
       foreach ($uids as $uid) {
         preg_match('#\((.*?)\)#', $uid, $match);
         $users_array .= $match[1] . ',';
       }
-      $user_default_page_config_entity->setUsers($users_array);
     }
+    $user_default_page_config_entity->setUsers($users_array);
     $status = $user_default_page_config_entity->save();
 
     switch ($status) {
